src/WingmanDataTrail/ListControls/QuickSearch/CountsProvider/CountsProvider.ts

```diff
@@ -0,0 +1,24 @@
+import { SceneObjectBase, type SceneObjectState } from '@grafana/scenes';
+
+export type CountsData = {
+  current: number;
+  total: number;
+};
+
+export interface CountsProviderState extends SceneObjectState {
+  counts: CountsData;
+}
+
+export class CountsProvider<T extends CountsProviderState = CountsProviderState> extends SceneObjectBase<T> {
+  constructor(state: Partial<T>) {
+    super({
+      ...state,
+      key: state.key || 'CountsProvider',
```

Comment:
Because we're instantiating the `MetricVariableCountsProvider` in a couple of places and using this `state.key || 'CountsProvider'`, I suggest that we either:
- Remove the `|| 'CountsProvider'` so that [`@grafana/scenes` can create a UUID](https://github.com/grafana/scenes/blob/d11f6e4728b5829a5b6121bc555d7e71588abb31/packages/scenes/src/core/SceneObjectBase.tsx#L44-L46), or
- Specify keys in each `new MetricVariableCountsProvider()`